XenModel/Actions/CrossConnectionParallelAction.cs

```diff
@@ -0,0 +1,158 @@
+﻿/* Copyright (c) Citrix Systems Inc. 
```

Comment:
In the new commit that I've added, I changed the ParallelAction to work across connections and got rid of the CCPA class.
I've created a separate ticked for making the ParallelAction (and MultipleAction) cancellable.